actors/market/src/lib.rs

```diff
@@ -1025,6 +979,81 @@ impl Actor {
             }
         }
     }
+
+    fn process_deal_updates(
+        rt: &impl Runtime,
+        params: ProcessDealsParams,
+    ) -> Result<ProcessDealsReturn, ActorError> {
+        let curr_epoch = rt.curr_epoch();
+        let mut total_slashed = TokenAmount::zero();
+        let mut terminated_deals: Vec<DealID> = Vec::new();
```

Comment:
Separate this into expired (the first case, cleaned up) and terminated (the second case). Propagate this to the return value too.

Reply:
Just to be clear on terminology. You want to differentiate between

_expired_: 
- cleaned up when `get_active_deal_or_cleanup` was called as it missed start_epoch

_terminated_:
- slashed deals
- deals that have completed past their end_epoch (these are already referred to as 'expired' in the existing code)

We can discuss better names to avoid overloading "expired"

Reply:
Ah, there are three different cases to distinguish then. But I definitely wanted to distinguish slashed deals from the others.

Reply:
If terminations are processed sycnhronously, they'll never happen here. We'll only have expired (unactivated) and completed.